src/pages/RFOX/hooks/useStakingInfoQuery.ts

```diff
@@ -0,0 +1,62 @@
+import { skipToken } from '@tanstack/react-query'
+import { foxStakingV1Abi } from 'contracts/abis/FoxStakingV1'
+import { RFOX_PROXY_CONTRACT_ADDRESS } from 'contracts/constants'
+import { useMemo } from 'react'
+import type { Address, ReadContractReturnType } from 'viem'
+import { getAddress } from 'viem'
+import { readContract } from 'viem/actions'
+import { arbitrum } from 'viem/chains'
+import type { Config } from 'wagmi'
+import { type ReadContractQueryKey, useQuery } from 'wagmi/query'
+import { viemClientByNetworkId } from 'lib/viem-client'
+
+type StakingInfoQueryKey = ReadContractQueryKey<
+  typeof foxStakingV1Abi,
+  'stakingInfo',
+  readonly [Address],
+  Config
+>
+type StakingInfo = ReadContractReturnType<typeof foxStakingV1Abi, 'stakingInfo', readonly [Address]>
+type UseStakingInfoQueryProps<SelectData = StakingInfo> = {
+  stakingAssetAccountAddress: string | undefined
+  select?: (stakingInfo: StakingInfo) => SelectData
+}
+const client = viemClientByNetworkId[arbitrum.id]
+
+export const useStakingInfoQuery = <SelectData = StakingInfo>({
+  stakingAssetAccountAddress,
+  select,
+}: UseStakingInfoQueryProps<SelectData>) => {
+  // wagmi doesn't expose queryFn, so we reconstruct the queryKey and queryFn ourselves to leverage skipToken type safety
+  const queryKey: StakingInfoQueryKey = [
```

Comment:
is memoization needed here?

Reply:
Won't hurt to! Not a big deal if we don't as react-queries are effectively singletons in a way (something something shared cache listeners) and even though this will re-render everything down the line, said re-render is no-opy

f12e79d8ccfa1bb74653f9a5e1785372d6b5cf86